src/mcts/node.h

```diff
@@ -266,8 +292,10 @@ class Node {
   // Pointer to a parent node. nullptr for the root.
   Node* parent_ = nullptr;
   // Pointer to a first child. nullptr for a leaf node.
+  // As a 'hack' actually a unique_ptr to Node[] if solid_children.
   std::unique_ptr<Node> child_;
```

Comment:
Maybe make it `std::unique_ptr<Node[]>` (with just one element by default)?
That way there need to be no hacks, everything will will automatically work..

It probably may be possible even not to touch iteration, we'd just need to make sibling_ a raw pointer rather than unique_ptr and rename flag to `sigling_is_owned_`.

Reply:
I considered both of these. But I was cautious due to a couple of reasons.
1) IIUC Node[] allocations unlike allocator.allocate(n) can store the length of the allocation so that delete[] knows how many items to destroy. I was concerned that the 'default case' this additional length storage might reduce likelihood that memory for a single node will be cacheline aligned and also waste more ram. I haven't tested it.
2) I prefer the iteration code being a separate path - advancing a pointer is considerably cheaper than dereferencing a sibling chain (let alone including the index checks to ensure they line up with the edge index desired) even when the sibling chain points directly to the next element.

I'd prefer to investigate 1) as a followup.  It doesn't change the overall behavior with respect to lifetime of node pointers being valid, so more people can be testing whether I have missed any race conditions sooner.